src/NHibernate/Cache/StandardQueryCache.cs

```diff
@@ -346,23 +387,55 @@ public void Destroy()
 				}
 				else
 				{
-					cacheable.Add(TypeHelper.Disassemble((object[])row, returnTypes, null, session, null));
+					cacheable.Add(TypeHelper.Disassemble((object[]) row, returnTypes, null, session, null));
 				}
 			}
 
 			return cacheable;
 		}
 
+		private static long GetResultsMetadata(IList cacheable, out string[] aliases)
+		{
+			aliases = null;
+
+			var metadata = cacheable[0];
+			if (metadata is long timestamp)
+				return timestamp;
+
+			var metadataArray = (object[]) metadata;
+			aliases = ((object[]) metadataArray[1]).Cast<string>().ToArray();
```

Comment:
Would this work?
```suggestion
			aliases = ((IEnumerable) metadataArray[1]).ToArray<string>();
```

Reply:
There is no LINQ extensions on `IEnumerable` except casting one to generic one Cast/OfType

Reply:
I think it should work for `IEnumerable<object>`

Reply:
No, `ToArray` definition is `TSource[] ToArray<TSource>(this IEnumerable<TSource>)`, so only a `ToArray<object>` can compile without a cast.